storage/src/test/java/com/tencent/rss/storage/handler/impl/LocalFileHandlerTest.java

```diff
@@ -116,17 +121,39 @@ private void writeTestData(
 
   protected void validateResult(ServerReadHandler readHandler, Set<Long> expectedBlockIds,
       Map<Long, byte[]> expectedData) {
-    ShuffleDataResult sdr = readHandler.getShuffleData(0);
-    byte[] buffer = sdr.getData();
-    List<BufferSegment> bufferSegments = sdr.getBufferSegments();
+    List<ShuffleDataResult> shuffleDataResults = readAll(readHandler);
     Set<Long> actualBlockIds = Sets.newHashSet();
-    for (BufferSegment bs : bufferSegments) {
-      byte[] data = new byte[bs.getLength()];
-      System.arraycopy(buffer, bs.getOffset(), data, 0, bs.getLength());
-      assertEquals(bs.getCrc(), ChecksumUtils.getCrc32(data));
-      assertTrue(Arrays.equals(data, expectedData.get(bs.getBlockId())));
-      actualBlockIds.add(bs.getBlockId());
+    for (ShuffleDataResult sdr : shuffleDataResults) {
+      byte[] buffer = sdr.getData();
+      List<BufferSegment> bufferSegments = sdr.getBufferSegments();
+
+      for (BufferSegment bs : bufferSegments) {
+        byte[] data = new byte[bs.getLength()];
+        System.arraycopy(buffer, bs.getOffset(), data, 0, bs.getLength());
+        assertEquals(bs.getCrc(), ChecksumUtils.getCrc32(data));
+        assertArrayEquals(expectedData.get(bs.getBlockId()), data);
+        actualBlockIds.add(bs.getBlockId());
+      }
     }
     assertEquals(expectedBlockIds, actualBlockIds);
   }
+
```

Comment:
may be we need some case about readIndex in this test.

Reply:
Read index is already covered by read data procedure.

Reply:
Should we add some cases, index file is correct, but data file is corrupted.

Reply:
It seems not be resolved.